From 71ffef862895e52ccc205bc513ae3c39b87a9854 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 18 Mar 2014 18:22:12 -0400
Subject: [PATCH 0155/1936] Gcc 4.8 build fix

For some reason the old code would result in a segfault,
which this code fixes
---
 src/Params/FilterParams.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index 230c08a5..67948b85 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -32,6 +32,9 @@
 
 using namespace rtosc;
 
+// g++ 4.8 needs this variable saved separately, otherwise it segfaults
+constexpr int sizeof_pvowels = sizeof(FilterParams::Pvowels);
+
 #define rObject FilterParams::Pvowels_t::formants_t
 static rtosc::Ports subsubports = {
     rParam(freq, "Formant frequency"),
@@ -82,7 +85,7 @@ rtosc::Ports FilterParams::ports = {
     {"Pvowels", NULL, NULL,
         [](const char *, RtData &d) {
             FilterParams *obj = (FilterParams *) d.obj;
-            d.reply(d.loc, "b", sizeof(FilterParams::Pvowels), obj->Pvowels);
+            d.reply(d.loc, "b", sizeof_pvowels, obj->Pvowels);
         }},
 
     {"Pvowels#" STRINGIFY(FF_MAX_VOWELS) "/", NULL, &subports,
-- 
2.47.0

